build_scripts/api/dreambooth/shared.py

```
@@ -0,0 +1,439 @@
+        force_cpu, embeddings_dir, sd_model
+    try:
+        import modules.script_callbacks
+        from modules import shared as ws
+        from modules.paths import models_path as mp, script_path as sp, sd_path as sdp
+        models_path = mp
+        script_path = sp
+        ckpt_dir = ws.cmd_opts.ckpt_dir
+        device_id = ws.cmd_opts.device_id
+        CLIP_stop_at_last_layers = ws.opts.CLIP_stop_at_last_layers
+        disable_safe_unpickle = ws.cmd_opts.disable_safe_unpickle
+        dataset_filename_word_regex = ws.opts.dataset_filename_word_regex
+        dataset_filename_join_string = ws.opts.dataset_filename_join_string
+        show_progress_every_n_steps = ws.opts.show_progress_every_n_steps
+        parallel_processing_allowed = ws.parallel_processing_allowed
+        state = ws.state
+        ckptfix = ws.cmd_opts.ckptfix
+        profile_db = ws.cmd_opts.profile_db
+        debug = ws.cmd_opts.debug_db
+        medvram = ws.cmd_opts.medvram
+        lowvram = ws.cmd_opts.lowvram
+        config = ws.cmd_opts.config
+        device = ws.device
+        sd_model = ws.sd_model
+        in_progress = False
+        in_progress_epoch = 0
+        in_progress_step = 0
+
+        def set_model(new_model):
+            global sd_model
+            sd_model = new_model
+
+        # Keep a reference to loaded script
+        modules.script_callbacks.on_model_loaded(set_model)
+
+        try:
+            dreambooth_models_path = ws.cmd_opts.dreambooth_models_path or dreambooth_models_path
+            ui_lora_models_path = ws.cmd_opts.lora_models_path or ui_lora_models_path
+            embeddings_dir = ws.cmd_opts.embeddings_dir or embeddings_dir
+        except:
+            pass
+
+        try:
+            force_cpu = ws.cmd_opts.force_cpu
+            if force_cpu:
+                device = torch.device("cpu")
+        except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```